build.py

```diff
@@ -23,18 +27,263 @@ def wait_until_container_is_ready():
+    return False
+
+def oc_create_service_account(project, service_account, privledge):
+    subprocess.run(['oc', r'-n', project, 'create', 'sa',service_account])
+    subprocess.run(['oc', 'adm', 'policy', r'add-cluster-role-to-user', privledge, r'system:serviceaccount:' + project + r':' + service_account])
+    return True
+
+
+def oc_project_exists(project):
+    result=subprocess.run(['oc', 'get', 'project', project],stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+    if(compare_results(result,r'^'+project)):
+        return True
+    return False
+
+def oc_create_project(project):
+    subprocess.run(['oc','new-project',project])
+    return True
+
+def oc_quota_exists(project,quota_name):
+    result=subprocess.run(['oc', 'describe', 'quota' , quota_name],stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+    if(compare_results(result,r'^'+quota_name)):
+        return True
+    return False
+
+def oc_create_quota(project,quota_name):
+    result=subprocess.run(['oc','create','-n',project,'-f', 'compute_resource.json'],stdout=subprocess.PIPE,stderr=subprocess.PIPE,stdin=subprocess.PIPE)
+    if(compare_results(result,r'^'+quota_name)):
+        return True
+    return False
+
+    #proc=subprocess.Popen(['oc','create','-n',project,'-f','-'],stdout=subprocess.PIPE,stderr=subprocess.PIPE,stdin=subprocess.PIPE)
+    #json_str=get_quota_def(project,quota_name)
+    #stdout_value=proc.communicate(json_str.encode())
+    #print('stdout:', repr(stdout_value))
+
+
+def get_quota_def(project,quota_name):
+    quota_def='apiVersion: v1\n' \
+	 +'kind: ResourceQuota\n' \
+	 +'metadata: \n' \
+	 +'  name:' +quota_name+'\n' \
+	 +'spec: \n' \
+	 +'  hard: \n' \
+	 +'    limits.cpu: 4\n' \
+	 +'    limits.memory: 2Gi\n' \
+	 +'    pods: 5\n' \
+	 +'    requests.cpu: 1\n' \
+	 +'    requests.memory: 1Gi\n'
+    return quota_def
+
+
+def oc_route_exists(project,route,host_subdomain):
+    result=subprocess.run(['oc', r'-n',project, 'get', 'route', route],stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+    if(compare_results(result,r'^'+route)):
+        return True
+    return False
+
+def oc_create_route(project,route,host_subdomain,service):
+    subprocess.run(['oc', r'-n',project, 'create', 'route', 'edge', r'--service=' + service, r'--hostname='+route+'.'+host_subdomain])
+    subprocess.run(['oc', r'-n',project, 'annotate', 'route',route, r'kubernetes.io/tls-acme=true'])
+    return True
+
+def oc_dc_exists(project,dc):
+    result=subprocess.run(['oc', r'-n',project, 'get', 'dc', dc],stdout=subprocess.PIPE,stderr=subprocess.STDOUT)
+    if(compare_results(result,r'^'+dc)):
+        return True
+    return False
+
+def get_dc_def(openshift_url,microserver_url,project,docker_image):
+    dc=""
+    if(openshift_url is not None and len(openshift_url)>0 and
+       microserver_url is not None and len(microserver_url)>0 and
+       project is not None and len(project)>0 and
+       docker_image is not None and len(docker_image)>0):
```

Comment:
It's not necessarily to check both for `var is not None` and `len(var) > 0`.  Both `None` and an empty string evaluate to `False` in a boolean context, so you can write:

```
if openshift_url and microserver_url and project and docker_image:
```

But if all those things aren't true, this returns an empty string to the caller. Is that useful? Or should this be raising a `ValueError` or otherwise reporting an error?